src/lib/list/list.ts

```diff
@@ -175,3 +204,122 @@ export class MdListItem implements AfterContentInit {
     return this._element.nativeElement;
   }
 }
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-list-option',
+  host: {
+    'role': 'option',
+    'class': 'mat-list-item',
+    '(focus)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(click)': 'onchange()',
+    '(keydown)':'onKeydown($event)',
+    '[tabIndex]': 'disabled ? -1 : 0',
+    '[attr.aria-selected]': '',
+
+  },
+  templateUrl: 'list-option.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class MdListOption implements AfterContentInit {
+  private _lineSetter: MdLineSetter;
+  private _disableRipple: boolean = false;
+  private _isNavList: boolean = false;
+  private _isSelectionList: boolean = false;
+  isSelected: boolean = false;
+
+  /**
+   * Whether the ripple effect on click should be disabled. This applies only to list items that are
+   * part of a nav list. The value of `disableRipple` on the `md-nav-list` overrides this flag.
+   */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  @ContentChildren(MdLine) _lines: QueryList<MdLine>;
+
+  /** Whether the label should appear after or before the checkbox. Defaults to 'after' */
+
+  @Input() checkboxPosition: 'before' | 'after' = 'after';
+
+  @ViewChild('autocheckbox') pCheckbox;
+
+  constructor(private _renderer: Renderer2,
+              private _element: ElementRef,
+              @Optional() private _slist: MdSelectionList,
+              @Optional() navList: MdNavListCssMatStyler,
+              @Optional() selectionListStyler: MdSelectionListCssMatStyler,
+              @Optional() public selectionList: MdSelectionListCheckboxer) {
+    this._isNavList = !!navList;
+    this._isSelectionList = !!selectionListStyler;
+  }
+
+
+  ngAfterContentInit() {
+    this._lineSetter = new MdLineSetter(this._lines, this._renderer, this._element);
+  }
+
+  onchange(): void {
```

Comment:
`onchange` -> `toggle`

It's generally better to name functions for what they do rather than based on how they are called. The exception is when the function behavior directly depends on specific event information (such as looking at which key was pressed)

The function also needs a JsDoc description